### PR TITLE
[BD-32] feat: add async feature to Hooks Extension Framework tooling

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description:**

This PR allows the [Hooks Extension Framework: Pipeline](https://github.com/edx/edx-django-utils/blob/457b307d1ee1d2d3329442811254657aa3f9f8ec/edx_django_utils/hooks/pipeline.py#L12) (check out its [ADR](https://github.com/edx/edx-django-utils/blob/457b307d1ee1d2d3329442811254657aa3f9f8ec/docs/decisions/0006-hooks-tooling-pipeline.rst)) to run asynchronously using celery tasks.

**JIRA:**

[XXX-XXXX](https://openedx.atlassian.net/browse/XXX-XXXX)

**Dependencies:**

This PR depends on [Hooks Extension Framework: Tooling](https://github.com/edx/edx-django-utils/pull/110)

**Merge deadline:**

None

**Testing instructions:**

TODO: make test repository usable for aysnc

**Reviewers:**
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:**

- [ ] Will there be a problem with adding celery? Adding it will cause other applications to crash?

### CODE REVIEW

File: edx_django_utils/hooks/tasks.py [hunk 1]
```diff
@@ -3,12 +3,15 @@
...
 
+from celery import shared_task
+
 from .exceptions import HookException
 from .utils import get_functions_for_pipeline
 
 log = getLogger(__name__)
 
 
+@shared_task()
```
Comment (mariajgrimaldi) [inline edx_django_utils/hooks/tasks.py:14]: Something interesting I found in edx-celeryutils was a BaseTask called [LoggedTask](https://github.com/edx/edx-celeryutils/blob/be8c68a72d144d720f28622f1963330c451b20f6/celery_utils/logged_task.py#L13), wouldn't be useful to have that extra logging?

File: edx_django_utils/hooks/tests/test_pipeline.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def setUp(self):
         }
         self.pipeline = Mock()
 
-    @patch("edx_django_utils.hooks.pipeline.get_functions_for_pipeline")
+    @patch("edx_django_utils.hooks.tasks.get_functions_for_pipeline")
```
Comment (mariajgrimaldi) [inline edx_django_utils/hooks/tests/test_pipeline.py:27]: here just changing the pipeline location from pipeline.py to tasks.py

File: edx_django_utils/hooks/triggers.py [hunk 3]
```diff
@@ -35,10 +41,17 @@ def trigger_filter(trigger_name, *args, **kwargs):
...
-        result = run_pipeline(
-            pipeline, *args, raise_exception=True, **kwargs
-        )  # TODO: change to async call.
+        try:
+            result = run_pipeline.delay(pipeline, *args, raise_exception=True, **kwargs)
+        except (TypeError, EncodeError):
+            log.exception(
+                "An error ocurred in trigger_filter while executing `run pipeline` with arguments: %s, %s.",
+                str(args),
+                str(kwargs),
```
Comment (mariajgrimaldi) [inline edx_django_utils/hooks/triggers.py:54]: maybe this won't be needed?

Comment (felipemontoya) [reply edx_django_utils/hooks/triggers.py:54]: It makes no difference what the pipeline is at this point right?
I mean, we are sure the pipeline object is never the source of this error. It is always the serialization, right?

Comment (mariajgrimaldi) [reply edx_django_utils/hooks/triggers.py:54]: Yes, it should always be serialization. I'll add it to the message!

**Comment by mariajgrimaldi** [comment]
The [discuss](https://discuss.openedx.org/t/configuration-for-the-hooks-extension-framework/4527) and [OEP](https://github.com/edx/open-edx-proposals/pull/184/) feedback has led us to remove native support for async. 

It is therefore preferable for events to follow the pattern:

> Listen on a signal and then create the async task

But that will be the responsibility of whoever writes the action.

**Pull request closed without merging**